### PR TITLE
Update query serializer, clean up, and bump version

### PR DESCRIPTION
**Pull request by soffes**
Code review:

- [x] @hyperspacemark 
- [x] @eliperkins

### CODE REVIEW

**Comment by hyperspacemark** [comment]
:green_apple:

File: VENCore.podspec [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
   s.platform     = :ios, '6.0'
-  s.source       = { :git => "https://github.com/venmo/VENCore.git",
-                     :tag => "v#{s.version}"
-  }
+  s.source       = { :git => 'https://github.com/venmo/VENCore.git',
+                     :tag => "v#{s.version}" }
   s.source_files = 'VENCore/**/*.{h,m}'
-  s.dependency 'CMDQueryStringSerialization', '~>0.2.0'
-  s.requires_arc = true
+  s.dependency 'CMDQueryStringSerialization'
```
Comment (eliperkins) [inline VENCore.podspec]: By removing the version here, we're saying that any version of `CMDQueryStringSerialization` should work with this repo. If a user declares `pod 'CMDQueryStringSerialization', '~> 0.2.0'` in their Podfile and include this latest version of VENCore that includes this change with no version restriction, there may be some issues in resolving the appropriate dependencies.

I'm really not for removing version requirements in podspecs, although it may be fine in Podfiles.

Comment (dasmer) [inline VENCore.podspec]: That's true. We only use the library in one place: https://github.com/venmo/VENCore/blob/master/VENCore/Networking/VENHTTP.m#L110

Maybe we can just use the implementation, so VENCore has 0 dependencies

Comment (soffes) [inline VENCore.podspec]: @eliperkins so the reason for this is I want to use version 0.3.0 in the Venmo app. I spoke with @calebd and he ensured me that the public API for serialization wouldn't change. To use 0.3.0, I had to remove the version restriction in `VENCore`, then update one of our private pods that had a version restriction on `VENCore` and then update all three in our app's Podfile. If we know this API won't change, I think it's best to be able to update it independently.

That said, @dasmer, if we only use it in one spot, maybe it's worth adding that specific category so we don't have an dependencies on something that should be super portable.

What do you guys think?

Comment (eliperkins) [inline VENCore.podspec]: Since this repo is public, and our iOS SDK depends on it, I'm :-1: in general on removing versions from dependencies to keep things simple for peeps who consume our stuff. If we wanna bring simple query parsing into this repo, that's cool with me, except that the iOS SDK is gonna use it on one line too: https://github.com/venmo/venmo-ios-sdk/blob/master/venmo-sdk/Venmo.m#L336

Don't feel too passionate on removing the dependency, especially since we only use it twice in this path. I _do_ feel pretty passionately about destroying dependency version resolution :P

Comment (dasmer) [inline VENCore.podspec]: I'm kind of OK either way. It's a hassle to update all the Pods with every version bump. 
Also, technically since the major version is 0, there can be breaking changes in any version updates.

@calebd if the public API is finalized and permanent, any reason not to release `1.0.0`? :octocat:

Comment (soffes) [inline VENCore.podspec]: Thinking about it more, I think it's a bad idea to remove @calebd's library if we need query string parsing. This junk is complicated and he has done a fanatic job handling all of the edge cases and writing tests for it. I definitely don't want to maintain that.

Totally down to lock to a version. If we need to update it (assuming we won't often), we can just bump the version of all of our junk.

**Comment by soffes** [comment]
Good to go, @eliperkins & @dasmer?

**Comment by soffes** [comment]
I broke travis. Standby

**Comment by soffes** [comment]
Good to go. To test:

``` sh
$ git clone https://github.com/venmo/VENCore
$ cd VENCore
$ bundle install
$ rake
#=> ... Executed 142 tests, with 0 failures (0 unexpected) in 0.978 (1.052) seconds
```

I'm a fan of this because it's easy to run tests the same way Travis does. You can also run `rake coverage` :)
